Money rework (3d model/texture)

Pyth3rEx opened the pull request:
Redone the 3d model and texture of the money stack to a single 1$ bill. #184 

Just a first little merge to get my hands into the project.

Todo: make a bigger stack for items above 1$ (eg. 3$, 15$, 50$ and 100$ stack 3d models)

Pyth3rEx committed:
Money Rework

Redone the 3d monel and texture to a single lower-res 1$ bill

Pyth3rEx committed:
3d model size fix

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/EveronLife/EveronLife?pullRequest=201) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/EveronLife/EveronLife?pullRequest=201) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/EveronLife/EveronLife?pullRequest=201) it.</sub>

thebonbon commented:
Please use .tif format for any images.
Also is it intended that the front and back are different sizes and front has a non-transparent empty space around it?

Pyth3rEx commented:
Wilco for .tif 

I guess the UV problem could be reworked quick. That was mostly an old 3d object I had laying around.

Pyth3rEx committed:
Redone everything

Redone 3d model, UV maps and textures to follow standards & conventions as well as limit clutter and file sizes.

Pyth3rEx committed:
Texture enhancements

Added displacement and normal maps, teaked settings for better texture quality

Pyth3rEx commented:
Fixed issues mentioned 

> Please use .tif format for any images. Also is it intended that the front and back are different sizes and front has a non-transparent empty space around it?

thebonbon reviewed (changes requested):
1. The cash texture has empty space in the middle.
2. The texture is 6.55MB. For an item that small it should be around or under 100KB. It does not need to be 1024x1024

Pyth3rEx committed:
fixed for PR

Removed blank space in texture
Lowered resolution to 256*213

thebonbon commented:
Looks good but not sure why the money looks green in the inventory. In first person it looks fine.
![image](https://user-images.githubusercontent.com/45827986/231521348-985d8243-5383-4ba9-b200-1c1a074c914d.png)

Pyth3rEx commented:
> Looks good but not sure why the money looks green in the inventory. In first person it looks fine. ![image](https://user-images.githubusercontent.com/45827986/231521348-985d8243-5383-4ba9-b200-1c1a074c914d.png)

I got no clue how the inventory processes the objects to make these icons. The green is a result of the displacement/normal map reshading. It's entended to look like that on the ground (you'll notice a thin translucent layer giving the green emulating the filigram on the real bill).

Is it possible to import our own icons to place in the inventory? If that's the case I could make a custom one and have it look good.

thebonbon commented:
> Is it possible to import our own icons to place in the inventory? If that's the case I could make a custom one and have it look good.
No it uses the 3d model. Guess the inventory does not like the translucent layer

Pyth3rEx committed:
Merge branch 'main' into main

Arkensor commented:
Just a quick note the smallest stack should still be a handful. It looks rather awkward if you have one thin paper bill in your hand. We have no animations that would look like that. The model size was already a good fit overall.

Pyth3rEx commented:
> Just a quick note the smallest stack should still be a handful. It looks rather awkward if you have one thin paper bill in your hand. We have no animations that would look like that. The model size was already a good fit overall. This green variant looks way too comic like IMO. It should be more something like this ![image](https://user-images.githubusercontent.com/8494013/232010809-7c8a8cdf-a673-4fe3-b875-335f05531377.png)

I think we should still have a single bill for when you have 1 dollar. I can make a mini stack for more than one bill ect (see to-do in initial post).

I will rework the colors as well, too bad I don't have an actual dollar to check against irl 😅

Pyth3rEx committed:
Merge pull request #1 from EveronLife/main

Updating following the persistence update

thebonbon commented on src/Assets/Items/Currencies/Data/cashTexture.tif:
Should be CashTexture_BCR.tif

> The texture suffix is important as the Workbench decides of a default import profile from it, greatly improving performances and storage space - see [Texture Properties](https://community.bistudio.com/wiki/Arma_Reforger:Textures#Texture_Properties) for a list of possible values.

thebonbon replied:
See also
https://github.com/EveronLife/EveronLife/blob/main/.github/CONVENTIONS.md#file-suffixes

thebonbon commented on src/Assets/Items/Currencies/Data/cashTexture_displacement.tif:
Instead of displacement + normal map, I think a single NMO texture should be enough, Not sure if displacement is even used in Reforger

thebonbon commented on src/Assets/Items/Currencies/Data/material.emat:
Should not be named just "material". CashMaterial or something

thebonbon commented on src/Assets/Items/Currencies/Data/material.emat:
Texture should not be green in inventory.

jasper-one commented:
Hey @Pyth3rEx just wanted to see if you are able to follow this up and get it upto date with main while reflecting the requested changes above? 

All good if not, will close this branch as cleaning PRs right now.
